[OBSDEF-5242] added crds

alexander-rassanov opened the pull request:
## Purpose
[OBSDEF-5242](https://jira.cec.lab.emc.com:8443/browse/OBSDEF-5242)
Added CRDS

## PR checklist
- [ ] make test passed
- [ ] make build passed
- [ ] helm install <chart> passed
- [ ] vSphere7 make package deployments passed
- [ ] helm upgrade <chart> passed
- [ ] helm test <release_name> passed

## Testing
_Paste the output of your helm install/vSphere7 deployment testing here_

eddiepavkovic commented on objectscale-manager/crds/storagepolicy.yaml:
```diff
@@ -0,0 +1,25 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: storagepolicies.ecs.dellemc.com
```
we need to rename this,  it conflicts with storagepolicies on vsphere.

```
─ $ ▶  kubectl get storagepolicies
NAME                                    AGE
dellemc-objectscale-highly-available    13h
dellemc-objectscale-vsan-direct-thick   13h
dellemc-objectscale-vsan-sna-thick      13h

```

alexander-rassanov replied:
fixed

eddiepavkovic replied:
but this name is still storagepolicies...  which kinda conflicts with vsphere storagepolicies...  
```suggestion
  name: storagepolicies.ecs.dellemc.com
```

alexander-rassanov replied:
However we have group ECS

alexander-rassanov replied:
is it ok for you?

eddiepavkovic commented on objectscale-manager/crds/storagepolicy.yaml:
```diff
@@ -0,0 +1,25 @@
+    kind: StoragePolicy
+    listKind: StoragePolicyList
+    plural: ecs-storagepolicies
+    singular: ecs-storagepolicy
```
jason wants us to not use `ecs` in the charts.  can we use `objectscale-storagepolicies`
```suggestion
    singular: ecs-storagepolicy
```

alexander-rassanov replied:
fixed

eddiepavkovic reviewed (dismissed):
Alex i brought this up during our standup and they agreed we shouldn't be re-using a name already taken by VMware.  can we prefix this with something more unique or come up with a different name.

alexander-rassanov commented:
> Alex i brought this up during our standup and they agreed we shouldn't be re-using a name already taken by VMware. can we prefix this with something more unique or come up with a different name.

I changed the name and added prefix objectstore. The common name of the CRD is under dell.ecs group
